### ISSUE TITLE
Warning reported for assignment to out parameter with [MaybeNull] or [MaybeNullWhen(...)]

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#39922]
No warning should be reported for the assignments:
```C#
#nullable enable

using System.Diagnostics.CodeAnalysis;

class Program
{
    static void GetValue<T>([AllowNull]T x, [MaybeNull]out T y)
    {
        y = x; // warning CS8601: Possible null reference assignment
    }

    static bool TryGetValue<T>([AllowNull]T x, [MaybeNullWhen(false)]out T y)
    {
        y = x; // warning CS8601: Possible null reference assignment
        return y != null;
    }
}
```

For `[MaybeNullWhen(...)]`, perhaps the compiler should allow assignment without warning to the variable regardless of the return value.

cc @stephentoub

**Comment by jcouv** [comment]
Relates to https://github.com/dotnet/roslyn/issues/36039 (Should nullability attributes affect method bodies and OHI?)
